mlrun/datastore/targets.py

```diff
@@ -1151,7 +1151,7 @@ def write_dataframe(
         else:
             # To prevent modification of the original dataframe and make sure
             # that the last event of a key is the one being persisted
-            df = df.groupby(df.index).last()
+            df = df.groupby(df.index.names).last()
```

Comment:
What if there is no index? `df.index.names` will be `[None]`. I doubt that will do anything good.

Reply:
Fixed